### ISSUE TITLE
Feature: add files outside user directory

### ISSUE THREAD

**Issue by bab1ch** [Cube777/dotgit#9]
It would be useful if one could add files outside user directory. There are a lot of files in `/etc/` that I would like to add to repository.

The simplest way for a user would be to add files with absolute path to `filelist`, although I would find it acceptable to have separate file or command for managing external files.

**Comment by lamarpavel** [comment]
There is an implied problem here. Files outside of your home directory are often read-only for *other* users, dotgit would have to deal with not being able to replace those with symlinks. And just to be clear, files in /etc should probably not be replaced by symlinks to your home directory since other users and groups will not be able to read them then.

Maybe dotgit could add a restriction to only include files of the same owner as the dotgit repository and a system-wide dotgit-repo should then be created by root and added somewhere in /usr/local or /var.

**Comment by Cube777** [comment]
@lamarpavel - thank you for pointing out these problems. Adding support for absolute paths shouldn't be a big problem. I don't think that the /etc files should be mixed with your home files though, @bab1ch - as @lamarpavel suggested, you could maybe create a dotgit repo for your root user and version /etc files from there - separate from your day to day dotfiles. Keep in mind that your dotfiles directory could be theoretically anywhere on your system, and as @lamarpavel pointed out, a location such as /var could be a good place to store them (anywhere world-readable should be good).

Please note that anything permission sensitive is going to be a problem - git disregards file permissions by default and currently dotgit cannot preserve file permissions other than locally. 

There is also the possibility of not using symlinks at all - you can create copies of your /etc folder by using the "hard" versions of the dotgit commands (check help). This way you can keep the repo inside your home folder and only update the repo from time to time and use root only when you want to restore the dotfiles should it be necessary. I believe the hard-copy method to be the cleanest and easiest with the only drawback being that you manually have to update the repo (using the `dotgit update` command) whenever you want to add the changes to the repo.

**Comment by Cube777** [comment]
Hi @bab1ch ,

I finally got around to start implementing the absolute filepaths but decided against it as I was going along for the following reasons:

- You will probably almost always need a separate repo to deal with the fact that permissions outside of your home folder would be different (as @lamarpavel mentioned before). One of dotgit's most distinct features is the fact that you only need one repo - this feature is thus not in line with the current project goals
- Dotgit was written mainly for managing dotfiles, and they are almost exclusively found inside your home folder. With this in mind the whole script was written to accommodate files only inside your home folder. As I began implementing it I realised that I (unintentionally) deeply ingrained this idea in the programming and this rather trivial feature would require a substantial amount of changes to the code which could possibly introduce numerous bugs to a currently stable code-base.
- There are already dedicated tools designed with the idea of versioning files like /etc config files (for example [etckeeper](https://etckeeper.branchable.com/). These tools will probably do the job of versioning /etc files much better than dotgit as they are designed for this purpos.

I'm sorry that I made commitments so quickly but I believe this to be the better route for the project, thank you for your contribution.
